templates/terraform/constants/router.go.erb

```diff
@@ -0,0 +1,31 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2019 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+// customizeDiff func for additional checks on google_compute_router properties:
+func resourceComputeRouterCustomDiff(diff *schema.ResourceDiff, meta interface{}) error {
+
+	block := diff.Get("bgp.0").(map[string]interface{})
+	advertiseMode := block["advertise_mode"]
+	advertisedGroups := block["advertised_groups"].([]interface{})
+	advertisedIPRanges := block["advertised_ip_ranges"].([]interface{})
+
+	if advertiseMode == "DEFAULT" && len(advertisedGroups) != 0 {
+		return fmt.Errorf("Error in bgp: advertised_groups cannot be specified when using advertise_mode DEFAULT")
```

Comment:
I wanted to use errors.New() here but I couldn't figure out how to add a new import in the final combined go file.

Reply:
we run goimports as part of the magician, so you shouldn't have to worry about adding it explicitly. However, we use fmt.Errorf all the time so I wouldn't worry too much about it. Are there any specific advantages of errors.New that make you prefer it?

Reply:
Ok. I'll leave it as fmt.Errorf. I usually use fmt.Errorf when I need to format string in the error message and errors.New when i don't.